amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -76,47 +58,58 @@ public void testMultiStreamDeserializationFail() {
-        assertEquals(Optional.empty(), single.streamCreationEpochOptional());
-        assertActualStreamIdentifierExpected(arn, single);
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void testInstanceWithoutEpochOrArn() {
-        mockStatic(StreamARNUtil.class);
-        when(getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID))
-                .thenReturn(Optional.empty());
-
-        try {
-            StreamIdentifier.singleStreamInstance(DEFAULT_ARN.toString());
-        } finally {
-            verifyStatic(StreamARNUtil.class);
-            getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID);
+    public void testMultiStreamArnFail() {
+        for (final String pattern : Arrays.asList(
+                "arn:aws:kinesis::123456789012:stream/stream-name", // missing region
+                "arn:aws:kinesis:region::stream/stream-name", // missing account id
+                "arn:aws:kinesis:region:123456789:stream/stream-name", // account id not 12 digits
+                "arn:aws:kinesis:region:123456789abc:stream/stream-name", // 12char alphanumeric account id
+                "arn:aws:kinesis:region:123456789012:stream/" // missing stream-name
+        )) {
+            try {
+                StreamIdentifier.multiStreamInstance(Arn.fromString(pattern), EPOCH);
+                Assert.fail(pattern + " should not have created a StreamIdentifier");
+            } catch (final IllegalArgumentException iae) {
+                // expected; ignore
+            }
         }
     }
 
     @Test
-    public void testSingleStreamInstanceWithName() {
-        StreamIdentifier actualStreamIdentifier = StreamIdentifier.singleStreamInstance(STREAM_NAME);
+    public void testSingleStreamInstanceFromArn() {
+        final StreamIdentifier actualStreamIdentifier = StreamIdentifier.singleStreamInstance(DEFAULT_ARN);
+
+        assertTrue(actualStreamIdentifier.streamARNOptional().isPresent());
+        assertTrue(actualStreamIdentifier.accountIdOptional().isPresent());
         assertFalse(actualStreamIdentifier.streamCreationEpochOptional().isPresent());
-        assertFalse(actualStreamIdentifier.accountIdOptional().isPresent());
-        assertFalse(actualStreamIdentifier.streamARNOptional().isPresent());
-        assertEquals(STREAM_NAME, actualStreamIdentifier.streamName());
+
+        assertActualStreamIdentifierExpected(DEFAULT_ARN, actualStreamIdentifier);
+        assertEquals(actualStreamIdentifier.streamName(), actualStreamIdentifier.serialize());
+    }
+
+    @Test
+    public void testMultiStreamInstanceFromArn() {
+        final StreamIdentifier actualStreamIdentifier = StreamIdentifier.multiStreamInstance(DEFAULT_ARN, EPOCH);
+
+        assertTrue(actualStreamIdentifier.streamARNOptional().isPresent());
+        assertTrue(actualStreamIdentifier.accountIdOptional().isPresent());
+        assertTrue(actualStreamIdentifier.streamCreationEpochOptional().isPresent());
```

Comment:
Replace this epoch `assertTrue` with `assertEquals(EPOCH, ...)`. That is, don't just validate existence; validate the correct value.